docs/console/console-extensibility/registration.md

```diff
@@ -0,0 +1,236 @@
+- `permissions`: indicate which permissions users will need to have to see the extension once activated (further details in the [next section](#how-to-restrict-the-extension-usage))
+- `routes` (_required_): provide the menu items that should appear on the sidebar once the extension is activated
+<!-- TODO: We should explain that we can only put one menu item per extension and multiple extensions need to be created to put more than one in different locations -->
+
+### How to restrict the extension usage?
+
+A registered extensions can specifies an array of `permissions` that is used to check if a user can see the active extension in the Console or not. In particular, the user must have **at least one of the required permissions in the array**, and the permissions that can be registered on an extension are those of the Console listed in the table of the [Identity and access management page](../../development_suite/identity-and-access-management/console-levels-and-permission-management.md#identity-capabilities-inside-console).
+
+### How to configure correctly the extension menu item?
+
+![registration example main route](./images/registrationExampleMainRoute.png)
+
+![registration example category route](./images/registrationExampleCategoryRoute.png)
+
+The routes should contain the info to render the extension menu item on the Console sidebar, that  allow to access the extension. A route can be registered leaving `renderType` unsetted to be rendered as simple menu item or it can be a `category` route corresponding to a new custom menu group where attach menu items.
+
+:::info
+Each extension should have at least a route (that is not `category`) to have sense and could also add at most another `category` type route to add a custom group menu item
+:::
+
+A route, therefore, requires to provide the following information:
+- `id` (_required_): assign a unique id that can be used for applying overrides with activation or for attaching other menu items if the route renderType is `category`
+- `locationId` (_required_): choose a location to place your route (see the [supported locations](./locations.md))
+- `renderType`: select `category` if you want to add a new menu group, otherwise leave it unsetted to add a menu item
+- `parentId`: insert the route `id` of a menu group where you want to attach the menu item. It is possible to choose existing parentIds documented on the [supported locations page](./locations.md) or add a route of type `category` and use its `id`. This value can be omitted for the `category` route.
+- `destinationPath` (_required_): indicate the destination suffix to which you will be redirected when clicking the menu item and it will compose the URL where the iframe will be mounted; in fact, the resulting URL will be composed accoring this pattern `<locationPath>/extensions/<extensionId><destinationPath>`. This information should be defined but is ignored for the `category` route.
```

Comment:
```suggestion
- `destinationPath` (_required_): indicate the destination suffix to which you will be redirected when clicking the menu item and it will compose the URL where the iframe will be mounted; in fact, the resulting URL will be composed according to this pattern `<locationPath>/extensions/<extensionId><destinationPath>`. This information should be defined but is ignored for the `category` route.
```